#! /bin/sh /usr/share/dpatch/dpatch-run
## 117-fix-optimizer-bug-in-gphot.dpatch by  <kmccarty@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch by Harald Vogt <hvogt@ifh.de> to work around compiler
## DP: optimization problems in src/geant321/gphys/gphot.F.

@DPATCH@
diff -urNad cernlib-2005.05.09.dfsg~/src/geant321/gphys/gphot.F cernlib-2005.05.09.dfsg/src/geant321/gphys/gphot.F
--- cernlib-2005.05.09.dfsg~/src/geant321/gphys/gphot.F	1995-10-24 06:21:29.000000000 -0400
+++ cernlib-2005.05.09.dfsg/src/geant321/gphys/gphot.F	2005-12-07 15:01:52.558184613 -0500
@@ -240,7 +240,7 @@
                   ELSE
 C Radiative shell decay
                      JS = JFN+1+2*NSHELL+ISHELL
-                     JS = JPHFN+Q(JS)
+                     JS = JPHFN + INT (Q(JS))    ! compiler optimiztion problem H. Vogt 2004/04/29
                      NPOINT = Q(JS)
                      DO 40 I = 1,NPOINT
                         IF(RN05.LT.Q(JS+I)) THEN
@@ -296,7 +296,7 @@
                   ELSE
 c Nonradiative decay
                      JS = JFN+1+3*NSHELL+ISHELL
-                     JS = JPHFN+Q(JS)
+                     JS = JPHFN + INT (Q(JS))    ! compiler optimiztion problem H. Vogt 2004/04/29
                      NPOINT = Q(JS)
                      DO 60 I = 1,NPOINT
                         IF(RN05.LT.Q(JS+I)) THEN
